libcloudforensics/providers/aws/forensics.py

```diff
@@ -487,10 +488,47 @@ def InstanceNetworkQuarantine(
   try:
     aws_account = account.AWSAccount(zone)
     vpc = aws_account.ec2.GetInstanceById(instance_id).vpc
+    logger.info('Creating isolation security group')
     sg_id = \
       aws_account.ec2.CreateIsolationSecurityGroup(vpc, exempted_src_subnets)
+    logger.info('Replacing attached security groups with isolation group')
     aws_account.ec2.SetInstanceSecurityGroup(instance_id, sg_id)
   except errors.ResourceNotFoundError as exception:
     raise errors.ResourceNotFoundError(
       'Cannot qurantine non-existent instance {0:s}: {1!s}'.format(instance_id,
         exception), __name__) from exception
+
+def InstanceProfileMitigator(
+    zone: str,
+    instance_id: str,
+    revoke_existing: bool = False
+    ) -> None:
+  """Remove an instance profile attachment from an instance. Also, optionally
+  revoke existing issued tokens for the profile.
+
+  Args:
+    zone (str): AWS Availability Zone the instance is in.
+    instance_id (str): The id (i-xxxxxx) of the virtual machine.
+    revoke_existing (bool): True to revoke existing tokens for the profile's
+      role. False otherwise.
+
+  Raises:
+    ResourceNotFoundError: If the instance cannot be found, or does not have a
+      profile attachment.
+  """
+  logger.info('Finding profile attachment')
+  aws_account = account.AWSAccount(zone)
+  assoc_id, profile = aws_account.ec2.GetInstanceProfileAttachment(instance_id)
+
+  if not profile or not assoc_id:
+    raise errors.ResourceNotFoundError(
+        'Instance not found or does not have a profile attachment: {0:s}'.
+        format(instance_id), __name__)
```

Comment:
Split the string instead of below the `.`?

Reply:
Indented the `format` line two more spaces for readability.